lib/wallet/src/Cardano/Wallet/DB.hs

```diff
@@ -405,17 +412,27 @@ data DBLayerCollection stm m s k = DBLayerCollection
 -- | Create a legacy 'DBLayer' from smaller database layers.
 mkDBLayerFromParts
     :: forall stm m s k. (MonadIO stm, MonadFail stm)
-    => DBLayerCollection stm m s k -> DBLayer m s k
-mkDBLayerFromParts DBLayerCollection{..} = DBLayer
+    => TimeInterpreter IO
+    -> DBLayerCollection stm m s k
+    -> DBLayer m s k
+mkDBLayerFromParts ti DBLayerCollection{..} = DBLayer
     { initializeWallet = initializeWallet_ dbWallets
     , removeWallet = removeWallet_ dbWallets
     , listWallets = listWallets_ dbWallets
     , walletsDB = walletsDB_ dbCheckpoints
     , putCheckpoint = putCheckpoint_ dbCheckpoints
-    , readCheckpoint = readCheckpoint_ dbCheckpoints
+    , readCheckpoint = readCheckpoint'
     , listCheckpoints = listCheckpoints_ dbCheckpoints
     , putWalletMeta = putWalletMeta_ dbWalletMeta
-    , readWalletMeta = readWalletMeta_ dbWalletMeta
+    , readWalletMeta = \wid -> do
+        readCheckpoint' wid >>= \case
+            Nothing -> pure Nothing
+            Just cp -> do
+                currentEpoch <- liftIO $
+                    interpretQuery ti (epochOf $ cp ^. #currentTip . #slotNo)
+                del <- readDelegation_ (dbDelegation wid) currentEpoch
```

Comment:
I'm wondering whether we can get rid of the new `TimeInterpreter IO` argument by passing `(epochOf $ cp ^. #currentTip . #slotNo)` directly to `readDelegation_`, so that `readDelegation_` performs the conversion to `currentEpoch` internally, rather than being given the result of the conversion. 🤔 — But this can be done in a later pull request, this code is already good. 😊